Initial implementation

pbrisbin opened the pull request:
See `README.md`

pbrisbin committed:
Initial implementation

See `README.md`

pbrisbin commented on src/Rcl/Run.hs:
```diff
@@ -0,0 +1,81 @@
+      That version -> pure $ Added version
+      These fromVersion toVersion ->
+        Changed fromVersion toVersion
+          <$> getChangelogFromTo cpName toVersion fromVersion
```
Need to rename the function or accept the arguments in reverse order.

eborden commented on src/Rcl/Options.hs:
```diff
@@ -0,0 +1,35 @@
+  { oDebug :: Bool
+  , oFromResolver :: Resolver
+  , oToResolver :: Resolver
+  , oDependencies :: Set PackageName
```
I would be awesome if you could use the `stack.yaml` to source `package.yaml` files and pull dependencies from there. Shouldn't be too hard to do.

pbrisbin replied:
I did start with that intention, but as soon as you need to be generalized or flexible to possible project setups, it quickly spirals. It was *much* easier to leave it to the caller to do something like this: https://github.com/freckle/megarepo/blob/master/backend/scripts/dependencies

pbrisbin replied:
I could add this small shell snippet to the `README` though.

eborden commented on src/Rcl/ChangedPackage.hs:
```diff
@@ -0,0 +1,63 @@
+getResolverDiff :: Resolver -> Resolver -> RIO App [ChangedPackage]
+getResolverDiff fromResolver toResolver = do
+  req <-
+    parseRequestThrow
```
Very convenient that this exposes JSON.

eborden commented on src/Rcl/Run.hs:
```diff
@@ -0,0 +1,81 @@
+          <$> getChangelogFromTo cpName toVersion fromVersion
+
+    logDebug $ "Changed " <> displayShow cpName <> ": " <> displayShow change
+    pure $ renderChanged cpName change
```
I'd love if the output included a link to the stackage diff page.

pbrisbin committed:
Include Stackage diff URL

pbrisbin committed:
Use blockquote, not code tags

This will still visually separate each package's changelog, but actually
render any markdown present therein.

pbrisbin committed:
RecordWildcards

pbrisbin committed:
Update README example

pbrisbin committed:
Tweak argument names

pbrisbin committed:
Configure Restyled
